From 007dec6874490a399057a0cfff979f2e0278c708 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Tue, 19 Mar 2024 18:40:01 +0100
Subject: [PATCH 1075/1141] misc: stm32_omm: disable OMM clock at runtime

OMM clock should not be enabled at runtime. Remove the dependency
between OMM driver and its children.

Change-Id: I3e046d0d89a08e0b1facd1a4f88b2e7eb1a4b0ce
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/368090
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/misc/stm32_omm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/stm32_omm.c b/drivers/misc/stm32_omm.c
index 3a4639e10d15..ed8f3307859b 100644
--- a/drivers/misc/stm32_omm.c
+++ b/drivers/misc/stm32_omm.c
@@ -392,7 +392,7 @@ static int stm32_omm_probe(struct platform_device *pdev)
 		if (!child_access[i] || !of_device_is_available(omm->child_node[i]))
 			continue;
 
-		vdev = of_platform_device_create(omm->child_node[i], NULL, dev);
+		vdev = of_platform_device_create(omm->child_node[i], NULL, NULL);
 		if (!vdev) {
 			dev_err(dev, "Failed to create Octo Memory Manager child\n");
 			for (j = i; j > 0; --j) {
-- 
2.39.2

